portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java

```diff
@@ -39,10 +57,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Query query = new Query("Message").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
- 
+
+    // Get comment limit and check validity.
+    int commentLimit = getCommentLimit(request);
+    if (commentLimit == -1) {
```

Comment:
I guess this check is duplicated to the check in getCommentLimit.

Reply:
response.setContentType("text/html");
      response.getWriter().println("Please enter a non-negative integer");
      return;
    }
I want to return some text if the user input is invalid, which I'll later display in an alert window to notify the user. Is there a better way to achieve this? It does seem a bit repetitive now tho.

Reply:
how about 
```
int commentLimit = Integer.parseInt(getParameterWithDefault(request, "limit", "0"));
if (commentLimit < 0) {
 ...
}
```

Reply:
I think this will work, then I don't need the getCommentLimit() function anymore. thank you!